src/components/Header/Header.jsx

```diff
@@ -7,7 +7,7 @@ import InputBase from '@mui/material/InputBase';
 import client from '../../utils/client';
 import { useNavigate } from 'react-router-dom';
 
-const Header = ({ role }) => {
+const Header = ({ role, userId }) => {
```

Comment:
If you are loading userId from local storage, do you need to define it as a prop as well?